src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -29,24 +31,35 @@
 @RequestMapping("/decoder")
 @RequiredArgsConstructor
 public class DecoderController {
+
   public static final String UNKNOWN_ERROR_MESSAGE = "Unknown error in DecoderController";
 
   private final DecoderDaoImpl decoderDao;
   private final DeviceDaoImpl deviceService;
   private final DecoderMapper decoderMapper;
   private final StreamMapper streamMapper;
   private final DecoderService decoderService;
+  private final MaintainDeviceStatus maintainDeviceStatus;
 
   @GetMapping
   public List<DecoderDto> retrieveAllDecoders() {
     List<DecoderEntity> decoderEntity = decoderDao.getDecoders();
+    maintainDeviceStatus.maintainStatusField(decoderEntity);
     return decoderMapper.toDecoderDtos(decoderEntity);
   }
 
   @GetMapping("/{serialNumber}")
   public ResponseEntity<DecoderDto> retrieveDecoder(@PathVariable @Valid String serialNumber) {
-    return decoderDao
-        .findDecoder(serialNumber)
+    // maintain status field and create a log if status changed
+    Optional<DecoderDto> decoder = decoderDao.findDecoder(serialNumber);
+    if (decoder.isPresent()) {
+      List<DecoderEntity> decodersListTemp = new LinkedList<>();
+      decodersListTemp.add(decoderMapper.toDecoderEntity(decoder.get()));
+      maintainDeviceStatus.maintainStatusField(decodersListTemp);
+    }
+
+    // return decoder
+    return decoder
```

Comment:
I agree with @mohhef here. I think it would be simpler if the device/encoder/decoder object have some status field that gets changed at the device/encoder/decoder level, in our case the sample sender and sample receiver, which they would be responsible to notify the backend using some status change http request. Like that we could have some aspect to monitor said http request method to generate logs on the status.

Reply:
I have extracted the logging logic to an Aspect class.